backend/api/student/notifications.py

```
@@ -0,0 +1,310 @@
+            "id": invitation.professor.id,
+            "full_name": invitation.professor.full_name,
+            "email": invitation.professor.email
+        }
+    }
+
+@router.post("/invitations/{invitation_token}/respond")
+async def respond_to_invitation_by_token(
+    invitation_token: str,
+    response: InvitationResponse,
+    request,
+    db: Session = Depends(get_db)
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)